src/components/motor/client.cpp

```diff
@@ -0,0 +1,174 @@
+#include <components/motor/client.hpp>
+
+#include <algorithm>
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <utility>
+
+#include <common/v1/common.pb.h>
+#include <component/motor/v1/motor.grpc.pb.h>
+
+#include <common/utils.hpp>
+#include <components/motor/motor.hpp>
+#include <config/resource.hpp>
+#include <robot/client.hpp>
+
+namespace viam {
+namespace cppsdk {
+
+void MotorClient::set_power(double power_pct) {
+    viam::component::motor::v1::SetPowerRequest request;
+    viam::component::motor::v1::SetPowerResponse response;
+
+    grpc::ClientContext ctx;
+
+    *request.mutable_name() = this->name();
+    request.set_power_pct(power_pct);
+
+    grpc::Status status = stub_->SetPower(&ctx, request, &response);
+    if (!status.ok()) {
+        throw std::runtime_error(status.error_message());
+    }
+}
+
+void MotorClient::go_for(double rpm, double revolutions) {
+    viam::component::motor::v1::GoForRequest request;
+    viam::component::motor::v1::GoForResponse response;
+
+    grpc::ClientContext ctx;
+
+    *request.mutable_name() = this->name();
+    request.set_rpm(rpm);
+    request.set_revolutions(revolutions);
+
+    grpc::Status status = stub_->GoFor(&ctx, request, &response);
+    if (!status.ok()) {
+        throw std::runtime_error(status.error_message());
```

Comment:
This should be cleaned up and given a proper error type as part of RSDK-2758 Proper and consistent error handling & exception scheme